pylint/test/extensions/test_check_mccabe.py

```diff
@@ -0,0 +1,61 @@
+"""Tests for the pylint checker in :mod:`pylint.extensions.check_mccabe
+"""
+
+import os.path as osp
+import unittest
+
+from pylint import checkers
+from pylint.extensions.check_mccabe import McCabeMethodChecker
+from pylint.lint import PyLinter
+from pylint.reporters import BaseReporter
+
+
+class TestReporter(BaseReporter):
+
+    def handle_message(self, msg):
+        self.messages.append(msg)
+
+    def on_set_current_module(self, module, filepath):
+        self.messages = []
+
+
+class TestMcCabeMethodChecker(unittest.TestCase):
+    """Test McCabe Method Checker"""
+
+    expected_msgs = [
+        'f1 is too complex. The McCabe rating is 1',
+        'f2 is too complex. The McCabe rating is 1',
+        'f3 is too complex. The McCabe rating is 3',
+        'f4 is too complex. The McCabe rating is 2',
+        'f5 is too complex. The McCabe rating is 2',
+        'f6 is too complex. The McCabe rating is 2',
+        'f7 is too complex. The McCabe rating is 3',
+        'f8 is too complex. The McCabe rating is 4',
+        'f9 is too complex. The McCabe rating is 9',
+        'f10 is too complex. The McCabe rating is 11',
+        'method1 is too complex. The McCabe rating is 1',
+        'method2 is too complex. The McCabe rating is 18',
+        "'For 188' is too complex. The McCabe rating is 4",
```

Comment:
The test is [pylint/test/extensions/data/mccabe.py#L187](https://github.com/vauxoo-dev/pylint/blob/49991e53f611ddb246766ff0fafd80aeaf255140/pylint/test/extensions/data/mccabe.py#L187)

The code is 
```python
for count in range(10):
    if count == 1:
        exit(0)
    elif count == 2:
        exit(1)
    else:
        exit(2)
```

The four sentence is too complex, but don't have a name (like as methods), then mccabe use: `type sentence` plus line number to identify it
`For 188` is the sentence `for` in the line number 188 that have a mccabe rating of 4 this section (because don't have a parent method node, is a `for` isolated.)